### ISSUE TITLE
Clicking autocomplete options doesn't work anymore

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#10172]
It picks the one that's highlighted, not the one you clicked

**Comment by t3chguy** [comment]
bit of debug, this is because when you have a selected completion, clicking anywhere other than the input itself will cause a `set_selection` change, which causes the autocompleter to close before the onClick on it gets triggered

**Comment by bwindels** [comment]
Indeed, from my investigations on Friday: on editor blur (caused by clicking the autocomplete), onChange changes the state and rerender triggers a prop change for query in the autocomplete because the content has changed for the query that got replaced by previously hitting arrow up/down. This ends up hiding the autocomplete in processCompletions because the new query doesn't have any matches anymore (it's an emoji for emojis, seems to be " " for pills?). This all happens before the click event for the completion item can fire, as blur is fired first.

I managed to detect and bail out from this case in `Autocomplete::complete` for emojis (e.g. if last bit of query === selectedCompletion), but this won't work for pills.

I just successfully reproduced this on riot 0.17.9, the last pre-1.0 release, so this is not a recent regression. Given that this will be fixed automatically with https://github.com/vector-im/riot-web/issues/9866, that it has taken us this long to notice the regression, and that it looks like to fix it I'd need to start digging deep into slate, I'll stop working on this and de-prioritize it.
